File: android/src/main/java/com/azify/theme/Theme.java
package com.azify.theme;

import androidx.annotation.Nullable;

import com.azify.R;
import com.azify.Config;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReadableMap;
import com.facetec.sdk.FaceTecSDK;

public class Theme {
  private final Color color;
  private final General general;
  private final Image image;
  private final Frame frame;
  private final Guidance guidance;
  private final Oval oval;
  private final Feedback feedback;
  private final ResultScreen resultScreen;
  private final IdScan idScan;
  public static ReadableMap Style = null;

  public Theme(ReactApplicationContext context) {
    this.color = new Color();
    this.general = new General();
    this.frame = new Frame();
    this.guidance = new Guidance();
    this.oval = new Oval();
    this.feedback = new Feedback();
    this.resultScreen = new ResultScreen();
    this.idScan = new IdScan();
    this.image = new Image(context);
  }

  public static void setStyle(@Nullable ReadableMap style) {
    Style = style;
  }

  public static void setTheme() {
    if (Config.currentCustomization == null) {
      return;
    }

    Config.currentCustomization
      .getIdScanCustomization().customNFCStartingAnimation = R.drawable.facetec_nfc_starting_animation;
    Config.currentCustomization
      .getIdScanCustomization().customNFCScanningAnimation = R.drawable.facetec_nfc_scanning_animation;
    Config.currentCustomization
      .getIdScanCustomization().customNFCCardStartingAnimation = R.drawable.facetec_nfc_card_starting_animation;
    Config.currentCustomization
      .getIdScanCustomization().customNFCCardScanningAnimation = R.drawable.facetec_nfc_card_scanning_animation;

    Vocal.setVocalGuidanceSoundFiles();

    FaceTecSDK.setCustomization(Config.currentCustomization);
    FaceTecSDK.setLowLightCustomization(Config.currentCustomization);
    FaceTecSDK.setDynamicDimmingCustomization(Config.currentCustomization);
  }

  public int getColor(String key) {
    return this.color.getColor(key);
  }

  public int getImage(String key, int defaultImage) {
    return this.image.getImage(key, defaultImage);
  }

  public General getGeneral() {
    return this.general;
  }

  public Frame getFrame() {
    return this.frame;
  }

  public Guidance getGuidance() {
    return this.guidance;
  }

  public Oval getOval() {
    return this.oval;
  }

  public Feedback getFeedback() {
    return this.feedback;
  }

  public ResultScreen getResultScreen() {
    return this.resultScreen;
  }

  public IdScan getIdScan() {
    return this.idScan;
  }
}

File: ios/Theme/Theme.swift
//
//  Theme.swift
//  Pods
//
//  Created by Daniel Sansão Araldi on 15/08/25.
//

import FaceTecSDK
import Foundation
import UIKit

public class Theme {
  private let color: Color
  private let image: Image
  private let message: Message
  private let general: General
  private let frame: Frame
  private let guidance: Guidance
  private let oval: Oval
  private let feedback: Feedback
  private let resultScreen: ResultScreen
  private let idScan: IdScan
  public static var Style: NSDictionary?

  init() {
    self.color = Color()
    self.image = Image()
    self.message = Message()
    self.general = General()
    self.frame = Frame()
    self.guidance = Guidance()
    self.oval = Oval()
    self.feedback = Feedback()
    self.resultScreen = ResultScreen()
    self.idScan = IdScan()
  }

  @available(iOS 8.2, *)
  public class func setAppTheme(_ style: NSDictionary?) {
    Theme.Style = style

    Config.currentCustomization = getCustomizationForTheme()
    Config.currentLowLightCustomization = getLowLightCustomizationForTheme()
    Config.currentDynamicDimmingCustomization = getDynamicDimmingCustomizationForTheme()

    FaceTec.sdk.setCustomization(Config.currentCustomization)
    FaceTec.sdk.setLowLightCustomization(Config.currentLowLightCustomization)
    FaceTec.sdk.setDynamicDimmingCustomization(Config.currentDynamicDimmingCustomization)
  }

  @available(iOS 8.2, *)
  class func getCustomizationForTheme() -> FaceTecCustomization {
    var currentCustomization = FaceTecCustomization()
    currentCustomization = Config.retrieveConfigurationWizardCustomization()

    return currentCustomization
  }

  @available(iOS 8.2, *)
  class func getLowLightCustomizationForTheme() -> FaceTecCustomization {
    var currentLowLightCustomization: FaceTecCustomization = getCustomizationForTheme()
    currentLowLightCustomization = Config.retrieveLowLightConfigurationWizardCustomization()

    return currentLowLightCustomization
  }

  @available(iOS 8.2, *)
  class func getDynamicDimmingCustomizationForTheme() -> FaceTecCustomization {
    var currentDynamicDimmingCustomization: FaceTecCustomization = getCustomizationForTheme()
    currentDynamicDimmingCustomization =
      Config.retrieveDynamicDimmingConfigurationWizardCustomization()

    return currentDynamicDimmingCustomization
  }

  public func getAuthenticateMessage(_ key: String, defaultMessage: String) -> String {
    return self.message.getMessage("authenticateMessage", key: key, defaultMessage: defaultMessage)
  }

  public func getEnrollmentMessage(_ key: String, defaultMessage: String) -> String {
    return self.message.getMessage("enrollMessage", key: key, defaultMessage: defaultMessage)
  }

  public func getPhotoIDScanMessage(_ key: String, defaultMessage: String) -> String {
    return self.message.getMessage("scanMessage", key: key, defaultMessage: defaultMessage)
  }

  public func getPhotoIDScanMessage(_ parent: String, key: String, defaultMessage: String) -> String
  {
    return self.message.getMessage(
      "scanMessage", parent: parent, key: key, defaultMessage: defaultMessage)
  }

  public func getPhotoIDMatchMessage(_ key: String, defaultMessage: String) -> String {
    return self.message.getMessage("matchMessage", key: key, defaultMessage: defaultMessage)
  }

  public func getPhotoIDMatchMessage(_ parent: String, key: String, defaultMessage: String)
    -> String
  {
    return self.message.getMessage(
      "matchMessage", parent: parent, key: key, defaultMessage: defaultMessage)
  }

  public func getColor(_ key: String) -> UIColor {
    return self.color.getColor(key)
  }

  public func getImage(_ key: String, defaultImage: String) -> UIImage? {
    return self.image.getImage(key, defaultImage: defaultImage)
  }

  public func getGeneral() -> General {
    return self.general
  }

  public func getFrame() -> Frame {
    return self.frame
  }

  public func getGuidance() -> Guidance {
    return self.guidance
  }

  public func getOval() -> Oval {
    return self.oval
  }

  public func getFeedback() -> Feedback {
    return self.feedback
  }
  
  public func getResultScreen() -> ResultScreen {
    return self.resultScreen
  }
  
  public func getIdScan() -> IdScan {
    return self.idScan
  }
}
